ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java

```diff
@@ -471,4 +448,41 @@ public void setFqdn(String fqdn) {
     public void setRestAdapter(RestAdapter restAdapter) {
         this.restAdapter = restAdapter;
     }
+
+    @Override
+    public void postStart() {
+        super.postStart();
+        generatePassword();
+    }
+
+    /**
+     * Need to wait server to start before resetting the password.
+     */
+    protected void generatePassword() {
+        if (Strings.isBlank(getAttribute(AmbariServer.PASSWORD))) {
+            String password;
+            if (Strings.isBlank(getConfig(AmbariServer.PASSWORD))) {
```

Comment:
I'd change so that
String password = getConfig(AmbariServer.PASSWORD)
That way there is only a single call to getConfig